src/OperatingSystemApp/oscursormodel.js

```diff
@@ -0,0 +1,114 @@
+
+var _propFlags = GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT;
+
+var cursorThemes = [
+    'cursor-default',
+    'cursor-cheese',
+    'cursor-ice-cream',
+    'cursor-poop',
+    'cursor-magic-wand',
+    'cursor-disco',
+    'cursor-rainbow-unicorn',
+    'cursor-glitchy',
+    'cursor-daemon',
+];
+
+var OSCursorModel = GObject.registerClass({
+    Properties: {
+        theme: GObject.ParamSpec.string(
+            'theme', 'Cursor Theme', 'cursor-default',
+            _propFlags, null),
+        size: GObject.ParamSpec.int(
+            'size', 'Cursor Size', 'Supported sizes are 16 24 32 48 64 96 128 192 256',
+            _propFlags, 16, 256, 24),
+        speed: GObject.ParamSpec.double(
+            'speed', 'Cursor Speed', '',
+            _propFlags, -1.0, 1.0, 0.0),
+    },
+}, class OSCursorModel extends OSModel {
+    _init(props = {
+        schemaId: 'org.gnome.desktop.interface',
+        schemaPath: '/org/gnome/desktop/interface/',
+    }) {
+        this._theme = null;
+        this._cursorSettings = new Gio.Settings({
+            schemaId: 'org.gnome.desktop.peripherals.touchpad',
+            path: '/org/gnome/desktop/peripherals/touchpad/',
+        });
+
+        super._init(props);
+
+        /* Get resource mtime to know if we need to update icons or not */
+        try {
+            var gresource = Gio.File.new_for_path(
+                '/app/share/com.endlessm.HackToolbox/com.endlessm.HackToolbox.data.gresource');
+            var gresource_info = gresource.query_info('time::modified', 0, null);
+            this._gresource_mtime = gresource_info.get_attribute_uint64('time::modified');
+        } catch (e) {
+        }
+
+        this.bindSetting('cursor-size', 'size');
+        this.bindSetting('cursor-theme', 'theme');
+        this.bindSetting('speed', 'speed', this._cursorSettings);
+    }
+
+    reset() {
+        this.theme = 'cursor-default';
+        this.settings.reset('cursor-size');
+        this._cursorSettings.reset('speed');
+    }
+
+    get theme () {
+        return this._theme;
+    }
+
+    set theme (cursor_theme) {
+        if (!cursor_theme || this._theme === cursor_theme)
+            return;
+
+        this._theme = cursorThemes.indexOf(cursor_theme) < 0 ? 'cursor-default' : cursor_theme;
+
+        /* Build user custom cursors directory path */
+        var dir = Gio.File.new_for_path(GLib.build_pathv('/',
+            [GLib.get_home_dir(), '.icons', this._theme, 'cursors']));
+        var cursor_file = dir.get_child('left_ptr');
+        var cursor_info, cursor_mtime;
+
+        /* If cursor exits and its older than the resource file,
+         * then all we have to do is notify!
+         */
+        if (cursor_file.query_exists(null) &&
+            (cursor_info = cursor_file.query_info('time::modified', 0, null)) &&
+            (cursor_mtime = cursor_info.get_attribute_uint64('time::modified')) &&
+            cursor_mtime >= this._gresource_mtime) {
```

Comment:
I am not sure this mtime check really does what you intend it to do. The intention as I understand it is to make it so that a newer cursor is copied over only if we update the gresource asset. However, the timestamp of the gresource is managed by Flatpak/ostree and is always set to zero. (You will see it's from 12/31/1969 if you do `ls -l` on a deployed flatpak directory in the terminal.)

However, I don't think that it's a big deal to always copy the asset over when something changes, so perhaps we can just do that.